From d996c4295e099d8c8066a0cfd8fd3402395387f8 Mon Sep 17 00:00:00 2001
From: "Lopez Esparza, Ricardo" <ricardo.lopez.esparza@intel.com>
Date: Wed, 6 Apr 2016 16:20:19 -0500
Subject: [PATCH 03/41] ov8858: CHT_MRD: Fix PLL configurations.

Currently there is a problem in the rear camera of MRD
where some distortion is introduced in the image every 5
seconds.
This problem is caused by a bad syncronization between
the sensor and MIPI PLLs.

Change-Id: I7708291babe3dacfc7e60c2ee01dfa1e2d88a10b
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-31785
Signed-off-by: Lopez Esparza, Ricardo <ricardo.lopez.esparza@intel.com>
Reviewed-on: https://android.intel.com:443/491474
Reviewed-on: https://android.intel.com:443/504745
---
 .../camera/drivers/media/i2c/ov8858.h              | 32 +++++++++++-----------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h
index b35b1f6..17362d8 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h
@@ -380,12 +380,12 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 	{OV8858_8BIT, 0x0103, 0x01}, /* software_reset */
 	{OV8858_8BIT, 0x0100, 0x00}, /* software_standby */
 	/* PLL settings */
-	{OV8858_8BIT, 0x0300, 0x05}, /* pll1_pre_div = /4 */
-	{OV8858_8BIT, 0x0302, 0xAF}, /* pll1_multiplier = 175 */
+	{OV8858_8BIT, 0x0300, 0x00}, /* pll1_pre_div = /1 */
+	{OV8858_8BIT, 0x0302, 0x27}, /* pll1_multiplier = 39 */
 	{OV8858_8BIT, 0x0303, 0x00}, /* pll1_divm = /(1 + 0) */
 	{OV8858_8BIT, 0x0304, 0x03}, /* pll1_div_mipi = /8 */
-	{OV8858_8BIT, 0x030B, 0x02}, /* pll2_pre_div = /2 */
-	{OV8858_8BIT, 0x030D, 0x4E}, /* pll2_r_divp = 78 */
+	{OV8858_8BIT, 0x030B, 0x00}, /* pll2_pre_div = /1 */
+	{OV8858_8BIT, 0x030D, 0x27}, /* pll2_r_divp = 39 */
 	{OV8858_8BIT, 0x030E, 0x00}, /* pll2_r_divs = /1 */
 	{OV8858_8BIT, 0x030F, 0x04}, /* pll2_r_divsp = /(1 + 4) */
 	/* pll2_pre_div0 = /1, pll2_r_divdac = /(1 + 1) */
@@ -628,8 +628,8 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 	{OV8858_8BIT, 0x3D85, 0x14}, /* OTP_REG85 */
 	{OV8858_8BIT, 0x3D8C, 0x73}, /* OTP_SETTING_STT_ADDRESS */
 	{OV8858_8BIT, 0x3D8D, 0xDE}, /* OTP_SETTING_STT_ADDRESS */
-	{OV8858_8BIT, 0x3F08, 0x10}, /* PSRAM control register */
-	{OV8858_8BIT, 0x3F0A, 0x80}, /* PSRAM control register */
+	{OV8858_8BIT, 0x3F08, 0x08}, /* PSRAM control register */
+	{OV8858_8BIT, 0x3F0A, 0x00}, /* PSRAM control register */
 
 	{OV8858_8BIT, 0x4000, 0xF1}, /* BLC CTRL00 = default */
 	{OV8858_8BIT, 0x4001, 0x00}, /* BLC CTRL01 */
@@ -674,7 +674,7 @@ static struct ov8858_reg ov8858_BasicSettings[] = {
 	{OV8858_8BIT, 0x481F, 0x32}, /* clk_prepare_min = 0x32 */
 	{OV8858_8BIT, 0x4825, 0x3A}, /* lpx_p_min = 0x3A */
 	{OV8858_8BIT, 0x4826, 0x40}, /* hs_prepare_min = 0x40 */
-	{OV8858_8BIT, 0x4837, 0x14}, /* pclk_period = 0x14 */
+	{OV8858_8BIT, 0x4837, 0x15}, /* pclk_period = 0x15 */
 	{OV8858_8BIT, 0x4850, 0x10}, /* LANE SEL01 */
 	{OV8858_8BIT, 0x4851, 0x32}, /* LANE SEL02 */
 
@@ -783,7 +783,7 @@ static const struct ov8858_reg ov8858_8M[] = {
 	{OV8858_8BIT, 0x3830, 0x06}, /* Unknown */
 	{OV8858_8BIT, 0x3836, 0x01}, /* Unknown */
 	{OV8858_8BIT, 0x3D85, 0x14}, /* OTP_REG85 */
-	{OV8858_8BIT, 0x3F08, 0x10}, /* PSRAM control register */
+	{OV8858_8BIT, 0x3F08, 0x08}, /* PSRAM control register */
 	{OV8858_8BIT, 0x4000, 0xF1}, /* BLC CTRL00 = default */
 	{OV8858_8BIT, 0x4001, 0x00}, /* BLC CTRL01 */
 	{OV8858_8BIT, 0x4002, 0x27}, /* BLC offset = 0x27 */
@@ -813,7 +813,7 @@ static const struct ov8858_reg ov8858_8M[] = {
 	{OV8858_8BIT, 0x403D, 0x04}, /* BLC CTRL3D */
 	{OV8858_8BIT, 0x4600, 0x01}, /* Unknown */
 	{OV8858_8BIT, 0x4601, 0x97}, /* Unknown */
-	{OV8858_8BIT, 0x4837, 0x14}, /* pclk_period = 0x14 */
+	{OV8858_8BIT, 0x4837, 0x15}, /* pclk_period = 0x15 */
 	{OV8858_TOK_TERM, 0, 0}
 };
 
@@ -845,7 +845,7 @@ static const struct ov8858_reg ov8858_3276x1848[] = {
 	{OV8858_8BIT, 0x3830, 0x06}, /* Unknown */
 	{OV8858_8BIT, 0x3836, 0x01}, /* Unknown */
 	{OV8858_8BIT, 0x3D85, 0x14}, /* OTP_REG85 */
-	{OV8858_8BIT, 0x3F08, 0x10}, /* PSRAM control register */
+	{OV8858_8BIT, 0x3F08, 0x08}, /* PSRAM control register */
 	{OV8858_8BIT, 0x4000, 0xF1}, /* BLC CTRL00 = default */
 	{OV8858_8BIT, 0x4001, 0x00}, /* BLC CTRL01 */
 	{OV8858_8BIT, 0x4002, 0x27}, /* BLC offset = 0x27 */
@@ -875,7 +875,7 @@ static const struct ov8858_reg ov8858_3276x1848[] = {
 	{OV8858_8BIT, 0x403D, 0x04}, /* BLC CTRL3D */
 	{OV8858_8BIT, 0x4600, 0x01}, /* Unknown */
 	{OV8858_8BIT, 0x4601, 0x97}, /* Unknown */
-	{OV8858_8BIT, 0x4837, 0x14}, /* pclk_period = 0x14 */
+	{OV8858_8BIT, 0x4837, 0x15}, /* pclk_period = 0x15 */
 	{OV8858_TOK_TERM, 0, 0}
 };
 
@@ -907,7 +907,7 @@ static const struct ov8858_reg ov8858_6M[] = {
 	{OV8858_8BIT, 0x3830, 0x06}, /* Unknown */
 	{OV8858_8BIT, 0x3836, 0x01}, /* Unknown */
 	{OV8858_8BIT, 0x3D85, 0x14}, /* OTP_REG85 */
-	{OV8858_8BIT, 0x3F08, 0x10}, /* PSRAM control register */
+	{OV8858_8BIT, 0x3F08, 0x08}, /* PSRAM control register */
 	{OV8858_8BIT, 0x4000, 0xF1}, /* BLC CTRL00 = default */
 	{OV8858_8BIT, 0x4001, 0x00}, /* BLC CTRL01 */
 	{OV8858_8BIT, 0x4002, 0x27}, /* BLC offset = 0x27 */
@@ -937,7 +937,7 @@ static const struct ov8858_reg ov8858_6M[] = {
 	{OV8858_8BIT, 0x403D, 0x04}, /* BLC CTRL3D */
 	{OV8858_8BIT, 0x4600, 0x01}, /* Unknown */
 	{OV8858_8BIT, 0x4601, 0x97}, /* Unknown */
-	{OV8858_8BIT, 0x4837, 0x14}, /* pclk_period = 0x14 */
+	{OV8858_8BIT, 0x4837, 0x15}, /* pclk_period = 0x15 */
 	{OV8858_TOK_TERM, 0, 0}
 };
 
@@ -1123,7 +1123,7 @@ static const struct ov8858_reg ov8858_1640x1232[] = {
 	{OV8858_8BIT, 0x403D, 0x04}, /* BLC CTRL3D */
 	{OV8858_8BIT, 0x4600, 0x00}, /* Unknown */
 	{OV8858_8BIT, 0x4601, 0xCB}, /* Unknown */
-	{OV8858_8BIT, 0x4837, 0x14}, /* pclk_period = 0x14 */
+	{OV8858_8BIT, 0x4837, 0x15}, /* pclk_period = 0x15 */
 	{OV8858_TOK_TERM, 0, 0}
 };
 
@@ -1185,7 +1185,7 @@ static const struct ov8858_reg ov8858_1640x1096[] = {
 	{OV8858_8BIT, 0x403D, 0x04}, /* BLC CTRL3D */
 	{OV8858_8BIT, 0x4600, 0x00}, /* Unknown */
 	{OV8858_8BIT, 0x4601, 0xCB}, /* Unknown */
-	{OV8858_8BIT, 0x4837, 0x14}, /* pclk_period = 0x14 */
+	{OV8858_8BIT, 0x4837, 0x15}, /* pclk_period = 0x15 */
 	{OV8858_TOK_TERM, 0, 0}
 };
 
@@ -1248,7 +1248,7 @@ static const struct ov8858_reg ov8858_1640x926[] = {
 	{OV8858_8BIT, 0x403D, 0x04}, /* BLC CTRL3D */
 	{OV8858_8BIT, 0x4600, 0x00}, /* Unknown */
 	{OV8858_8BIT, 0x4601, 0xCB}, /* Unknown */
-	{OV8858_8BIT, 0x4837, 0x14}, /* pclk_period = 0x14 */
+	{OV8858_8BIT, 0x4837, 0x15}, /* pclk_period = 0x15 */
 	{OV8858_TOK_TERM, 0, 0}
 };
 
-- 
1.9.1

